NautilusCode/program_list.py

```diff
@@ -34,9 +34,97 @@
                  Flatpak("com.sublimetext.three"),
                  Native("subl"))
 
+progs += Program('android-studio', _('Android Studio'),
+                 Native('studio'),
+                 Flatpak('com.google.AndroidStudio'))
+
+progs += Program('aqua', _('Aqua'),
+                 Native('aqua'))
+
+progs += Program('clion', _('CLion'),
+                 Native('clion'),
+                 Flatpak('com.jetbrains.CLion'),)
+
+progs += Program('clion-eap', _('CLion (EAP)'),
+                 Native('clion-eap'))
+
+progs += Program('datagrip', _('DataGrip'),
+                 Native('datagrip'),
+                 Flatpak('com.jetbrains.DataGrip'))
+
+progs += Program('datagrip-eap', _('DataGrip (EAP)'),
+                 Native('datagrip-eap'))
+
+progs += Program('dataspell', _('DataSpell'),
+                 Native('dataspell'))
+
+progs += Program('dataspell-eap', _('DataSpell (EAP)'),
+                 Native('dataspell-eap'))
+
+progs += Program('fleet', _('Fleet'),
+                 Native('fleet'))
+
+progs += Program('goland', _('GoLand'),
+                 Native('goland'),
+                 Flatpak('com.jetbrains.GoLand'))
+
+progs += Program('goland-eap', _('GoLand (EAP)'),
+                 Native('goland-eap'))
+
+# IntelliJ IDEA has multiple editions that default to the same command but
+# prioritise the highest-paying edition. Therefore the native version has been
+# treated as a separate program due to the uncertainty around edition.
+progs += Program('idea', _('IntelliJ IDEA'),
+                 Native('idea'))
+
+progs += Program('idea-eap', _('IntelliJ IDEA (EAP)'),
+                 Native('idea-eap'))
+
+progs += Program('idea-community', _('IntelliJ IDEA Community'),
+                 Flatpak('com.jetbrains.IntelliJ-IDEA-Community'))
+
+progs += Program('idea-professional', _('IntelliJ IDEA Ultimate'),
+                 Flatpak('com.jetbrains.IntelliJ-IDEA-Ultimate'))
+
+progs += Program('mps', _('MPS'),
+                 Native('mps'))
+
 progs += Program('phpstorm', _("PhpStorm"),
-                 Flatpak("com.jetbrains.PhpStorm"),
-                 Native("phpstorm"))
+                 Native("phpstorm"),
+                 Flatpak("com.jetbrains.PhpStorm"))
```

Comment:
This is an unrelated/unecessary change and should not be part of this MR.

Reply:
Is this not a part of adding support for all JetBrains IDEs by maintaining consistency between how they are coded (including it's alphabetical position in the code)?

I'll revert the change, however I didn't consider that to be a refactor but instead an implementation that includes all products instead of just the one.

Reply:
Well, nautilus-code already supports PhpStorm.

So, this change does not fit commit "**Add** JetBrains support".

A better way to do this would be to sort the whole list in a single commit outside this MR.